bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -471,7 +474,17 @@ private static NameCharacteristic createNameCharacteristic(HomekitTaggedItem tag
 
     private static HoldPositionCharacteristic createHoldPositionCharacteristic(HomekitTaggedItem taggedItem,
             HomekitAccessoryUpdater updater) {
-        return new HoldPositionCharacteristic(value -> ((SwitchItem) taggedItem.getItem()).send(OnOffType.from(value)));
+        return new HoldPositionCharacteristic(value -> {
+            if (!value) {
+                return;
+            }
+            Item item = taggedItem.getBaseItem();
+            if (item instanceof SwitchItem) {
+                ((SwitchItem) item).send(OnOffType.ON);
+            } else if (item instanceof RollershutterItem) {
```

Comment:
possible for item to be neither of these? Maybe we should add a final else close and log an error?

Reply:
good call. will do (but slightly differently, so the warning comes as soon as the accessory is created, not later when a Hold is actually received).